baseplate/diagnostics/tracing/__init__.py

```diff
@@ -494,3 +505,42 @@ def flush_func(self, spans):
             )
         except RequestException as e:
             self.logger.error("Error flushing spans: %s", e)
+
+
+class TraceTooLargeError(Exception):
+    pass
+
+
+class TraceQueueFullError(Exception):
+    pass
+
+
+MAX_SIDECAR_QUEUE_SIZE = 102400
+MAX_SIDECAR_MESSAGE_SIZE = 10000
+
+
+class SidecarRecorder(BaseBatchRecorder):
+    """Interface for recording spans to a POSIX message queue.
+
+    The SidecarRecorder deserializes spans to a string representation before
+    adding them to the queue.
+    """
+    def __init__(self, queue_name):
+        self.queue = MessageQueue(
+            "/traces-" + queue_name,
+            max_messages=MAX_SIDECAR_QUEUE_SIZE,
+            max_message_size=MAX_SIDECAR_MESSAGE_SIZE
+        )
+
+    def send(self, span):
+        # Don't raise exceptions from here. This is called in the
+        # request/response path and should finish cleanly.
+        serialized_str = json.dumps(span._serialize())
+        if len(serialized_str) > MAX_SIDECAR_MESSAGE_SIZE:
+            logger.error("Trace too big")
+        try:
```

Comment:
nit: could you put a blank between the if and the try so they don't read as one big control structure please?